src/main/java/com/google/sps/data/NSWGovAdaptor.java

```diff
@@ -32,6 +32,20 @@ public NSWGovAdaptor() throws Exception {
     return this.dataMap;
   }
 
+  private AQDataPoint convertDataPoint(NSWGovAQDataPoint givenPoint) { 
+    // Take the site id from given NSWGovAQDataPoint.
+    int pointSiteid = givenPoint.getId();
+
+    // Fill all data into AQdatapoint.
+    AQDataPoint siteDataPoint = new AQDataPoint();
+    siteDataPoint.lat = this.dataMap.get(pointSiteid).lat;
+    siteDataPoint.lng = this.dataMap.get(pointSiteid).lng;
```

Comment:
In some places, we have a Coordinates class, in others, we're re-creating local lat/long pairs. It would be better (in a subsequent CL), if all lat/long references were encapsulated in a Coordinates (with the exception of where that would break JSON translation).

Reply:
ah, okay. Initially I used getcoord and placed them into a coordinates class, but I was thinking if that was unncessary longer code since I was able to get it immediately from the map.

Reply:
Each of the map lookups is an additional data fetching operation of a higher order of complexity that directly accessing an already retrieved data object (and then has to do the local access as well).

Fetching it from the map once only incurs the map lookup a single time (one extra line of code), but then only has to use local data references to assign those values. Even though it's one line longer, it's much clearer to read, and also more efficient.

Shortest code is not always most readable code.